### DISCUSSION TITLE
.gitignore not ignoring the files specified.

### DISCUSSION THREAD

**Discussion by maarid** [community/community#24023] [New to GitHub]
<p>I have updated the .gitignore file with the entry /app/etc/config.php<br>
Then i pulled this .gitignore file onto my server<br>
Then when i change the app/etc/config.php file and push it to github, it still accepts and pushes it to repository!!!</p>
<p>Then i make a change in github directly on this file and do a pull on local system, it downloads this updated file!!!</p>
<p>What am i missing?</p>

**Comment by sfynx** [comment, marked as answer]
<p>A .gitignore file only affects untracked files that you don’t want to add to the repository.</p>
<p>It seems to me that /app/etc/config.php is already tracked in your repository. In that case pushing and pulling changes to it will not be affected by the .gitignore file, see also: <a href="https://git-scm.com/docs/gitignore" rel="nofollow noopener">https://git-scm.com/docs/gitignore</a></p>
<p>If you didn’t want to add this file to the repository, you can use <strong>git rm --cached</strong> to remove the file from the index, but keep it around locally.</p>

**Comment by JotSinghChawla** [comment]
<p>Hey Maarid,<br>
I have also faced this type of issue a while ago, if this issue is with less number of files what you can do is<br>
 → Copy the /app/etc/config.php locally<br>
 → Delete the file from the Github repository<br>
 → Pull the changes from Github<br>
 → And Copy the files ( stored locally ) again in a new file namely /app/etc/config.php</p>
<p>Make sure that you have added /app/etc/config.php in gitignore after pulling the changes from Github</p>
